File: core/src/main/java/com/gmail/marcosav2010/cipher/EncryptedMessage.java
package com.gmail.marcosav2010.cipher;

import com.gmail.marcosav2010.common.Utils;
import lombok.Getter;

public class EncryptedMessage {

    @Getter
    private final byte[] encryptedSimmetricKeyBytes, encryptedData, byteLength;

    EncryptedMessage(byte[] encryptedSimmetricKeyBytes, byte[] encryptedData) {
        this.encryptedSimmetricKeyBytes = encryptedSimmetricKeyBytes;
        this.encryptedData = encryptedData;
        byteLength = Utils.intToBytes(~intLength());
    }

    int intLength() {
        return encryptedData.length;
    }
}

File: core/src/main/java/com/gmail/marcosav2010/cipher/EncryptedMessage.kt
package com.gmail.marcosav2010.cipher

import com.gmail.marcosav2010.common.Utils.intToBytes

class EncryptedMessage internal constructor(val encryptedSymmetricKeyBytes: ByteArray, val encryptedData: ByteArray) {

    val byteLength = intToBytes(intLength().inv())

    private fun intLength() = encryptedData.size
}